crates/biome_css_parser/src/syntax/value/url.rs

```diff
@@ -0,0 +1,188 @@
+use crate::lexer::CssLexContext;
+use crate::parser::CssParser;
+
+use crate::syntax::value::parse_error::expected_url_modifier;
+use crate::syntax::{
+    is_at_any_function, is_at_identifier, is_at_string, is_nth_at_any_function,
+    parse_regular_identifier, parse_simple_function, parse_string,
+};
+use biome_css_syntax::CssSyntaxKind::*;
+use biome_css_syntax::{CssSyntaxKind, T};
+use biome_parser::parse_lists::ParseNodeList;
+use biome_parser::parse_recovery::{ParseRecovery, RecoveryResult};
+use biome_parser::parsed_syntax::ParsedSyntax;
+use biome_parser::parsed_syntax::ParsedSyntax::{Absent, Present};
+use biome_parser::{token_set, Parser, TokenSet};
+
+const URL_SET: TokenSet<CssSyntaxKind> = token_set![T![url], T![src]];
+
+/// Determines if the current position of the parser is at the beginning of a URL function.
+pub(crate) fn is_at_url_function(p: &mut CssParser) -> bool {
+    p.at_ts(URL_SET) && p.nth_at(1, T!['('])
+}
+
+/// Parses a URL function from the current position of the CSS parser.
+///
+/// This function is designed to parse URL functions as specified in the CSS Values and Units Module.
+/// The URL function can be of various forms and may include different types of content, like a standard
+/// URL, a URL with quotes, a URL with variables, or a URL with special characters. This parser handles
+/// these variations according to the syntax rules defined in the CSS specification.
+///
+/// For more detailed information on the CSS URL function syntax, refer to the
+/// [CSS Values and Units Module](https://drafts.csswg.org/css-values-4/#url-value).
+///
+/// # URL Function Syntax Examples
+///
+/// - Standard URL: `url(image.jpg)`
+///   Represents a basic URL function without any additional modifiers or quotes.
+///
+/// - URL with quotes: `url("image.jpg")`
+///   A URL function where the URL is enclosed within quotes.
+///
+/// - URL with a variable: `url(var(--image-path))`
+///   Demonstrates a URL function that uses a CSS variable as its value.
```

Comment:
I'm really wondering how we'll be able to handle this. Similar to `calc`, `var` can appear...almost anywhere lol. But it still has validation on what's allowed and what's not, so I figure maybe the parse tree will just have to have `CssMathFunctions` or `CssVarFunction` as alternatives for the typed value nodes like `CssLength` or `CssPercentage` and such.

Definitely not something for right now, though!